src/Neo.VM/ExecutionEngineLimits.cs

```diff
@@ -16,47 +16,103 @@ namespace Neo.VM
     /// <summary>
     /// Represents the restrictions on the VM.
     /// </summary>
-    public sealed record ExecutionEngineLimits
+    public sealed class ExecutionEngineLimits : IEquatable<ExecutionEngineLimits>
     {
         /// <summary>
         /// The default strategy.
         /// </summary>
-        public static readonly ExecutionEngineLimits Default = new();
+        public static readonly ExecutionEngineLimits Default = new ExecutionEngineLimits();
 
         /// <summary>
         /// The maximum number of bits that <see cref="OpCode.SHL"/> and <see cref="OpCode.SHR"/> can shift.
         /// </summary>
-        public int MaxShift { get; init; } = 256;
+        public int MaxShift
+        {
+            get;
+#if NET5_0_OR_GREATER
+            init;
+#else
+            set;
```

Comment:
Maybe create a constructor for all of these "else"? otherwise it can be changed after the creation